hack/bump-version.sh

```diff
@@ -26,23 +26,23 @@ if [ -z "$new_version" ]; then
 fi
 
 sed -i "s/$old_version/$new_version/g" cmd/kogito/version/version.go README.md version/version.go deploy/operator.yaml deploy/olm-catalog/kogito-operator/kogito-operator.package.yaml deploy/olm-catalog/kogito-operator/custom-subscription-example.yaml hack/go-build.sh hack/go-vet.sh .osdk-scorecard.yaml
-operator-sdk generate csv --apis-dir ./pkg/apis/apps/v1alpha1 --verbose --csv-version "$new_version" --from-version "$old_version" --update-crds --operator-name kogito-operator
 sed -i "s|operated-by: kogito-operator.*|operated-by: kogito-operator.${new_version}|g" deploy/olm-catalog/kogito-operator/manifests/kogito-operator.clusterserviceversion.yaml
 
+if [ "${old_version}" != "${new_version}" ]; then
+    operator-sdk generate csv --apis-dir ./pkg/apis/app/v1alpha1 --verbose --csv-version "$new_version" --from-version "$old_version" --update-crds --operator-name kogito-operator
+fi
+make vet
+
 # rewrite test default config, all other configuration into the file will be overridden
 test_config_file="test/.default_config"
-current_branch=`git branch --show-current`
-
-if [ "${release}" = "true" ]; then
-    image_version=`echo "${new_version}" | awk -F. '{print \$1"."\$2}'`
-    if [ "${no_release_branch}" = "true" ]; then
-        branch="master"
-    else
-        branch="${image_version}.x"
-    fi
-else
-    image_version="latest"
+
+image_version=`echo "${new_version}" | awk -F. '{print \$1"."\$2}'`
```

Comment:
refactored it for more flexibility.
Default, we consider being on a release branch when bumping the version.
Rest of the parameters would define any change